lib/intern.dl

```diff
@@ -0,0 +1,21 @@
+/* Object interning library.
+ * Generates a unique ID for an object, which can be converted back to the original value.
+ */
+
+/* Interned object of type `'A`.
+ * While this type is defined for any `'A`, interning is only supported for strings.
+ * There is simply no way to obtain an interned object of a different type.
+ */
+extern type IObj<'A>
+
+/* Interned string
+ */
+typedef IString = IObj<string>
+
+/* Intern a string
+ */
+extern function string_intern(s: string): IString
```

Comment:
I find these names rather baroque. Why not intern and get_string?

Reply:
I try to follow the "object oriented" naming convention: function names start with the name of the type the function applies to followed by a verb. If I call this just `intern`, what should I call similar methods for other (non-string) types?   Ideally, `intern()` should be polymorphic, but I haven't figured out a way to do it, since the interner is a global variable and we need one per every type that we want to intern.

Open to better suggestions, though.